linux/desktop_webview_auth_plugin.cc

```diff
@@ -0,0 +1,175 @@
+{
+  g_autoptr(FlMethodResponse) response = nullptr;
+
+  const gchar *method = fl_method_call_get_name(method_call);
+  FlValue *args = fl_method_call_get_args(method_call);
+
+  if (strcmp(method, "signIn") == 0)
+  {
+    FlView *view = fl_plugin_registrar_get_view(plugin->registrar);
+    GtkWindow *window = GTK_WINDOW(gtk_widget_get_toplevel(GTK_WIDGET(view)));
+    GtkWidget *popup = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+
+    gtk_window_set_transient_for(GTK_WINDOW(popup), GTK_WINDOW(window));
+
+    GtkWidget *scrollview = gtk_scrolled_window_new(nullptr, nullptr);
+
+    plugin->context = webkit_web_context_new ();
+    GtkWidget *webview = webkit_web_view_new_with_context (plugin->context);
+
+    gtk_container_add(GTK_CONTAINER(scrollview), webview);
+    gtk_container_add(GTK_CONTAINER(popup), scrollview);
+
+    FlValue *signInUriValue = fl_value_lookup_string(args, "signInUri");
+    const gchar *signInUri = fl_value_get_string(signInUriValue);
+
+    FlValue *redirectUrlValue = fl_value_lookup_string(args, "redirectUri");
+    const gchar *redirectUrl = fl_value_get_string(redirectUrlValue);
+
+    FlValue *widthString = fl_value_lookup_string(args, "width");
+    FlValue *heightString = fl_value_lookup_string(args, "height");
+    FlValue *dartNull = fl_value_new_null();
+
+    int width = 920;
+    int height = 720;
+
+    if (!fl_value_equal(widthString, dartNull))
+    {
+      width = fl_value_get_int(widthString);
+    }
+
+    if (!fl_value_equal(heightString, dartNull))
+    {
+      height = fl_value_get_int(heightString);
+    }
+
+    webkit_web_view_load_uri(WEBKIT_WEB_VIEW(webview), signInUri);
+
+    gtk_window_set_position(GTK_WINDOW(popup), GTK_WIN_POS_MOUSE);
+    gtk_window_set_default_size(GTK_WINDOW(popup), width, height);
+
+    plugin->redirectUrl = g_strdup(redirectUrl);
+    plugin->popupWindow = popup;
+
+    plugin->load_change_handler = g_signal_connect(G_OBJECT(webview), "load-changed", G_CALLBACK(changed), NULL);
+    g_signal_connect(G_OBJECT(popup), "destroy", G_CALLBACK(destroy), &plugin);
+
+    gtk_widget_show_all(popup);
+
+    response = FL_METHOD_RESPONSE(fl_method_success_response_new(nullptr));
+  }
+  else
+  {
+    response = FL_METHOD_RESPONSE(fl_method_not_implemented_response_new());
+  }
+
+  fl_method_call_respond(method_call, response, nullptr);
+}
+
+static void desktop_webview_auth_plugin_dispose(GObject *object)
+{
+  G_OBJECT_CLASS(desktop_webview_auth_plugin_parent_class)->dispose(object);
+}
+
+static void desktop_webview_auth_plugin_class_init(DesktopWebviewAuthPluginClass *klass)
+{
+  G_OBJECT_CLASS(klass)->dispose = desktop_webview_auth_plugin_dispose;
+}
+
+static void desktop_webview_auth_plugin_init(DesktopWebviewAuthPlugin *self) {}
+
+static void method_call_cb(FlMethodChannel *channel, FlMethodCall *method_call,
+                           gpointer user_data)
+{
+  desktop_webview_auth_plugin_handle_method_call(method_call);
```

Comment:
You can cast `user_data` to be able to pass the plugin instance to `desktop_webview_auth_plugin_handle_method_call` like in the original Flutter plugin template:

```cpp
DesktopWebbviewAuthPlugin* plugin = DESKTOP_WEBVIEW_AUTH_PLUGIN(user_data);
desktop_webview_auth_plugin_handle_method_call(plugin, method_call);
```